core/utils.py

```diff
@@ -1098,6 +1098,46 @@ def unescape_encoded_uri_component(escaped_string: str) -> str:
     return urllib.parse.unquote(escaped_string)
 
 
+def get_formatted_query_string(escaped_string: str) -> str:
+    """Returns a formatted query string that can be used to perform search
+    operations from escaped query string in url.
+
+    Args:
+        escaped_string: str. Query string that is encoded with
+            encodeURIComponent.
+
+    Returns:
+        str. Formatted query string which can be directly used to perform
+        search.
+    """
+    query_string = unescape_encoded_uri_component(escaped_string)
+    # Remove all punctuation from the query string, and replace it with
+    # spaces. See http://stackoverflow.com/a/266162 and
+    # http://stackoverflow.com/a/11693937
+    remove_punctuation_map = dict(
+        (ord(char), None) for char in string.punctuation)
+    query_string = query_string.translate(remove_punctuation_map)
+    return query_string
+
+
+def convert_filter_parameter_string_into_list(filter_string: str) -> List[str]:
+    """Converts the filter parameter string into a list of applied filter
+    values.Filter string should be in the following form: ("Algebra" OR "Math")
```

Comment:
```suggestion
    """Converts the filter parameter string into a list of applied filter
    values. Filter string should be in the following form: ("Algebra" OR "Math")
```

Also for the example, can there be more ORs?

Reply:
Added more examples. There can be multiple ORs in the string.